src/sqlfluff/core/templaters/slicers/tracer.py

```diff
@@ -139,16 +150,18 @@ def move_to_slice(self, target_slice_idx, target_slice_length):
                 candidates.sort(key=lambda c: abs(target_slice_idx - c))
                 self.program_counter = candidates[0]
 
-    def record_trace(self, target_slice_length):
```

Comment:
I tweaked this function to optionally take a `slice_idx` so I could reuse it to process output from a `{% set %}` trace. Previously it always used `self.program_counter`.